components/brave_component_updater/browser/BUILD.gn

```diff
@@ -1,9 +1,25 @@
 import("//extensions/buildflags/buildflags.gni")
 
+declare_args() {
+  updater_dev_endpoint = ""
+  updater_prod_endpoint = ""
+}
+
+config("updater_config") {
+  defines = [
+    "UPDATER_DEV_ENDPOINT=\"$updater_dev_endpoint\"",
+    "UPDATER_PROD_ENDPOINT=\"$updater_prod_endpoint\"",
+  ]
+}
+
 source_set("browser") {
   sources = [
     "brave_component.cc",
     "brave_component.h",
+    "brave_component_features.cc",
```

Comment:
Please follow established patterns: brave_component_features.cc -> features.cc and the same thing with switches.cc